src/engine/SCons/Scanner/Fortran.py

```diff
@@ -285,21 +285,26 @@ def FortranScan(path_variable="FORTRANPATH"):
 #   but *not* the following:
 #
 #   MODULE PROCEDURE procedure_name
+#   MODULE SUBROUTINE subroutine_name
+#   MODULE FUNCTION function_name
 #
 #   Here is a breakdown of the regex:
 #
-#   (?i)               : regex is case insensitive
-#   ^\s*               : any amount of white space
-#   MODULE             : match the string MODULE, case insensitive
-#   \s+                : match one or more white space characters
-#   (?!PROCEDURE)      : but *don't* match if the next word matches
-#                        PROCEDURE (negative lookahead assertion),
-#                        case insensitive
-#   (\w+)              : match one or more alphanumeric characters
-#                        that make up the defined module name and
-#                        save it in a group
-
-    def_regex = """(?i)^\s*MODULE\s+(?!PROCEDURE)(\w+)"""
+#   (?i)                               : regex is case insensitive
+#   ^\s*                               : any amount of white space
+#   MODULE                             : match the string MODULE, case
+#                                        insensitive
+#   \s+                                : match one or more white space
+#                                        characters
+#   (?!PROCEDURE|SUBROUTINE|FUNCTION)  : but *don't* match if the next word
+#                                        matches PROCEDURE, SUBROUTINE or
+#                                        FUNCTION (negative lookahead
+#                                        assertion), case insensitive
+#   (\w+)                              : match one or more alphanumeric
+#                                        characters that make up the defined
+#                                        module name and save it in a group
+
+    def_regex = """(?i)^\s*MODULE\s+(?!PROCEDURE|SUBROUTINE|FUNCTION)(\w+)"""
```

Comment:
If you're fiddling with the regexes anyway, please make them raw strings.  Python 3.8 is going to complain noisily otherwise - regex constructs containg `\c` where _c_ is meaningful to re will take a warning that they're not valid _Python_ escapes if it's not a raw string, like this:

`SyntaxWarning: invalid escape sequence \s`